src/databricks/labs/blueprint/paths.py

```diff
@@ -235,66 +277,318 @@ class WorkspacePath(Path):
+
+    @property
+    def name(self):
+        path_parts = self._path_parts
+        return path_parts[-1] if path_parts else ""
+
+    @property
+    def parts(self):
+        if self.drive or self.root:
+            return self.drive + self.root, *self._path_parts
+        return self._path_parts
+
+    @property
+    def suffix(self):
+        # Super implementations are mostly fine...
+        suffix = super().suffix
+        # ...but if there is no suffix and this path is for a notebook then infer the extension based on the notebook
+        # language.
+        if not suffix and self.is_notebook():
+            try:
+                suffix = self._SUFFIXES.get(self._object_info.language, "")
+            except DatabricksError:
+                pass
+        return suffix
+
+    @property
+    def suffixes(self):
+        # Super implementations are all fine.
+        # TODO: Make this consistent with .suffix for notebooks.
+        return super().suffixes
 
-    def _from_parts(self, args):  # pylint: disable=arguments-differ
-        # instance method adapted from pathlib.Path
-        drv, root, parts = self._parse_args(args)
-        return self._from_parsed_parts(drv, root, parts)
+    @property
+    def stem(self):
+        # Super implementations are all fine.
+        return super().stem
+
+    def with_name(self, name):
+        parser = self.parser
+        if not name or parser.sep in name or name == ".":
+            msg = f"Invalid name: {name!r}"
+            raise ValueError(msg)
+        path_parts = list(self._path_parts)
+        if not path_parts:
+            raise ValueError(f"{self!r} has an empty name")
+        path_parts[-1] = name
+        return type(self)(self._ws, self.anchor, *path_parts)
+
+    def with_stem(self, stem):  # pylint: disable=useless-parent-delegation
+        # Super implementations are all fine.
+        # TODO: Decide before PR merge whether to: a) remove; b) allow as marker that we checked it; c) inline to be independent.
+        return super().with_stem(stem)
+
+    def with_suffix(self, suffix):
+        stem = self.stem
+        if not stem:
+            msg = f"{self!r} has an empty name"
+            raise ValueError(msg)
+        if suffix and not suffix.startswith("."):
+            msg = f"{self!r} invalid suffix: {suffix}"
+            raise ValueError(msg)
+        return self.with_name(stem + suffix)
 
-    def relative_to(self, *other) -> pathlib.PurePath:  # type: ignore
-        """Databricks Workspace works only with absolute paths, so we make sure to
-        return pathlib.Path instead of WorkspacePath to avoid confusion."""
-        return pathlib.PurePath(super().relative_to(*other))
+    @property
+    def _stack(self):
+        return self.anchor, list(reversed(self._path_parts))
+
+    def relative_to(self, other, *more_other, walk_up=False):  # pylint: disable=arguments-differ
+        other = self.with_segments(other, *more_other)
```

Comment:
how much of this code is copy-pasted from py3.12 impl? we need to properly attribute larger chunks

Reply:
The implementation was heavily inspired by the (default) implementation from the ABC coming in python 3.13. I've since replaced it (and `is_relative_to()`) with an independent implementation.

In general there isn't cut'n'paste code, but the implementations are inspired by the PathLib versions:
 - They're usually simpler because we: a) don't deal with drives; b) assume posix-style paths; c) don't have to deal with case-folding.
 - Because the internal representation is different the implementations are also adapted accordingly. (In particular we store normalised path segments, which eliminates a lot of corner-cases for things like empty or `.` segments.)

Reply:
ok :) being inspired is sufficient.